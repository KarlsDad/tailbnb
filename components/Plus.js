import img1 from '../public/plus-1.jpg';

import PlusCard from './PlusCard';

export default () => {
  return (
    <div class='xl:max-w-layout lg:max-w-5.5xl mx-auto mt-14 mb-`20`'>
      <h3 className='text-2xl font-semibold py-1 mt-5 '>
        Airbnb Plus places to stay
      </h3>
      <p className='mb-6'>
        A selection of places to stay verified for quality and design.
      </p>
      <PlusCard img={img1} />
    </div>
  );
};
