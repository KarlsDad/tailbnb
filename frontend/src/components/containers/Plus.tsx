import * as React from 'react';

// Components
import { PlusCard } from '../presentational/PlusCard';

// Images
// High Resolution
const img1 = require('../../../public/img/high/plus-1.jpg');

export const Plus: React.FC<{}> = () => {
  return (
    <>
      <PlusCard img={img1} />
    </>
  );
};
