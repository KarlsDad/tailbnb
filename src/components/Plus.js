import React, { useState } from 'react';
import Skeleton from 'react-loading-skeleton';

// Components
import PlusCard from './PlusCard';

// Images
import img1 from '../../public/img/plus-1.jpg';

export default () => {
  const [loaded, setLoaded] = useState(false);

  const sleep = (milliseconds) => {
    return new Promise((resolve) => setTimeout(resolve, milliseconds));
  };

  const setLoad = () => {
    sleep(4000).then(() => {
      setLoaded(true);
    });
  };
  setLoad();
  return <>{loaded ? <PlusCard img={img1} /> : <Skeleton height={226} />}</>;
};
