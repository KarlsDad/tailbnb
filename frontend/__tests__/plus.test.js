import React from 'react';
import { shallow } from 'enzyme';

// Component
import Plus from '../src/components/Plus';

describe('Plus', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Plus />);
    expect(component).toMatchSnapshot();
  });
});
