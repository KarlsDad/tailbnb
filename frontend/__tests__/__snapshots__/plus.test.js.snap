// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plus should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
